nexus/src/app/sagas/region_replacement_start.rs

```diff
@@ -0,0 +1,1261 @@
+            params.request.id,
+            saga_id,
+            new_region_id,
+            old_region_volume_id,
+        )
+        .await
+        .map_err(ActionError::action_failed)?;
+
+    Ok(())
+}
+
+#[cfg(test)]
+pub(crate) mod test {
+    use crate::{
+        app::db::lookup::LookupPath, app::db::DataStore,
+        app::saga::create_saga_dag,
+        app::sagas::region_replacement_start::find_only_new_region,
+        app::sagas::region_replacement_start::Params,
+        app::sagas::region_replacement_start::SagaRegionReplacementStart,
+        app::sagas::test_helpers::test_opctx, app::RegionAllocationStrategy,
+    };
+    use chrono::Utc;
+    use nexus_db_model::Dataset;
+    use nexus_db_model::DatasetKind;
+    use nexus_db_model::Region;
+    use nexus_db_model::RegionReplacement;
+    use nexus_db_model::RegionReplacementState;
+    use nexus_db_model::Volume;
+    use nexus_db_queries::authn::saga::Serialized;
+    use nexus_db_queries::context::OpContext;
+    use nexus_test_utils::resource_helpers::create_disk;
+    use nexus_test_utils::resource_helpers::create_project;
+    use nexus_test_utils::resource_helpers::DiskTest;
+    use nexus_test_utils_macros::nexus_test;
+    use nexus_types::identity::Asset;
+    use sled_agent_client::types::VolumeConstructionRequest;
+    use uuid::Uuid;
+
+    type ControlPlaneTestContext =
+        nexus_test_utils::ControlPlaneTestContext<crate::Server>;
+
+    const DISK_NAME: &str = "my-disk";
+    const PROJECT_NAME: &str = "springfield-squidport";
+
+    #[nexus_test(server = crate::Server)]
+    async fn test_region_replacement_start_saga(
+        cptestctx: &ControlPlaneTestContext,
+    ) {
+        let mut disk_test = DiskTest::new(cptestctx).await;
+        disk_test.add_zpool_with_dataset(&cptestctx).await;
+
+        let client = &cptestctx.external_client;
+        let nexus = &cptestctx.server.server_context().nexus;
+        let datastore = nexus.datastore();
+        let opctx = test_opctx(cptestctx);
+
+        let _project_id =
+            create_project(&client, PROJECT_NAME).await.identity.id;
+
+        // Create a disk
+        let client = &cptestctx.external_client;
+        let disk = create_disk(&client, PROJECT_NAME, DISK_NAME).await;
+
+        // Assert disk has three allocated regions
+        let disk_id = disk.identity.id;
+        let (.., db_disk) = LookupPath::new(&opctx, &datastore)
+            .disk_id(disk_id)
+            .fetch()
+            .await
+            .unwrap_or_else(|_| panic!("test disk {:?} should exist", disk_id));
+
+        let allocated_regions =
+            datastore.get_allocated_regions(db_disk.volume_id).await.unwrap();
+        assert_eq!(allocated_regions.len(), 3);
+
+        // Replace one of the disk's regions
+        let region_to_replace: &nexus_db_model::Region =
+            &allocated_regions[0].1;
+
+        // Manually insert the replacement request
+        let request = RegionReplacement {
```

Comment:
Yeah, I've thought a bit about this. I'd like to do this work in a follow up PR though, as this would murder the other commits stacked on top of this one! I've opened https://github.com/oxidecomputer/omicron/issues/5858 and will follow up afterwards.

Reply:
> Yeah, I've thought a bit about this. I'd like to do this work in a follow up PR though, as this would murder the other commits stacked on top of this one! I've opened #5858 and will follow up afterwards.

I kinda figured as much. Definitely no rush. Thanks for opening an issue!